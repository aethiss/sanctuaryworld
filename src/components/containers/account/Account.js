import React from 'react'
import PropTypes from 'prop-types'

// Next
import Image from 'next/image'
import Link from 'next/link'

// Components
import BodyContainer from '../../commons/wrapper/BodyContainerResponsive'
import DiabloResponsiveButton from '../../commons/buttons/DiabloResponsiveButton/DiabloResponsiveButton'
import ButtonDescription from '../../commons/buttons/buttonDescription/ButtonDescription'

// Style
const styles = {
  container: {
    textAlign: 'center',
  },
  link: {
    textDecoration: 'underline',
  },
}

const AccountPage = ({ user }) => {
  const blizzardConnect = () => {
    return (
      <div style={styles.container}>
        <p>Secure Connect with Blizzard Account</p>
        <p>
          This is a secure connection using oAuth2 (
          <a href='https://oauth.net/2/' target='_blank'>
            <u>read more</u>
          </a>
          )
        </p>
        <DiabloResponsiveButton
          text='BATTLE.NET'
          anchorLink='/api/connect/bnet'
        />
      </div>
    )
  }

  return (
    <BodyContainer background='/images/background/act1-bg.jpg'>
      <div style={styles.container}>
        <Image
          src='/images/background/d4_logo.png'
          alt='diablo 4 logo'
          width={370}
          height={200}
        />
        {user.battletag ? (
          <p>Connected as {user.battletag} </p>
        ) : (
          blizzardConnect()
        )}
      </div>
    </BodyContainer>
  )
}

AccountPage.propTypes = {
  user: PropTypes.object,
}

AccountPage.defaultProps = {
  user: {},
}

export default AccountPage
