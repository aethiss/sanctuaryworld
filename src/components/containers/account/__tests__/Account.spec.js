import React from 'react'
import TestRenderer from 'react-test-renderer'
import Account from '../Account'

jest.mock('next/image', () => (jest.fn(() => null)))

const mockNotAuth = {
  user: {
    battletag: 'pippi#1123',
    auth: false,
  }
}

const mockAuth = {
  user: {
    battletag: 'pippi#1123',
    auth: 'sadfghjkuytrewqwerty',
  }
}

describe('Account', () => {
  describe('Render', () => {
    it('[Account] : Render default', () => {
      const testRenderer = TestRenderer.create(<Account />)
      expect(testRenderer).toMatchSnapshot()
    })
    it('[Account] : Render connected not authenticated', () => {
      const testRenderer = TestRenderer.create(<Account {...mockNotAuth} />)
      expect(testRenderer).toMatchSnapshot()
    })
    it('[Account] : Render connected and authenticated', () => {
      const testRenderer = TestRenderer.create(<Account {...mockAuth} />)
      expect(testRenderer).toMatchSnapshot()
    })
  })
})
