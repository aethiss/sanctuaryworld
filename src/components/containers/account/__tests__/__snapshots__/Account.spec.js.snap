// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Render [Account] : Render connected and authenticated 1`] = `
<div
  className="makeStyles-bodyWrapper-1"
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <span>
      <p>
        Connected as 
        pippi#1123
         
      </p>
      <div>
        <p>
          User Authenticated
        </p>
      </div>
    </span>
  </div>
</div>
`;

exports[`Account Render [Account] : Render connected not authenticated 1`] = `
<div
  className="makeStyles-bodyWrapper-1"
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <span>
      <p>
        Connected as 
        pippi#1123
         
      </p>
      <div>
        <div>
          <p>
            Sync your Blizzard Account with SanctuaryWorld !
          </p>
          <span>
            <a
              href="/login"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <button
                className="makeStyles-button-2"
              >
                <span
                  className="makeStyles-buttonText-3"
                >
                  Sign-In
                </span>
              </button>
            </a>
          </span>
        </div>
      </div>
    </span>
  </div>
</div>
`;

exports[`Account Render [Account] : Render default 1`] = `
<div
  className="makeStyles-bodyWrapper-1"
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <span>
      <div>
        <p>
          Secure Connect with Blizzard Account
        </p>
        <a
          href="/api/connect/bnet"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <button
            className="makeStyles-button-2"
          >
            <span
              className="makeStyles-buttonLeft-4"
            />
            <span
              className="makeStyles-buttonMid-3"
            >
              <span
                className="makeStyles-buttonText-6"
              >
                Connect
              </span>
            </span>
            <span
              className="makeStyles-buttonRight-5"
            />
          </button>
        </a>
        <p>
          Or 
          <a
            href="/login"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "textDecoration": "underline",
              }
            }
          >
            Login
          </a>
           with Sanctuary Account
        </p>
      </div>
       
      <br />
    </span>
  </div>
</div>
`;
